weldx/utility.py

```diff
@@ -562,6 +562,148 @@ def xr_interp_like(
     return result
 
 
+def xr_valid_key(dax: xr.DataArray, ref: dict):
+    """Validate if the keys in ref are present in the DataArray.
+
+    This is a helper function for xr_check_coords. Throws an exception when the
+    dimension is requested in dax and not optional.
+
+    Parameters
+    ----------
+    dax:
+        xarray object
+    ref:
+        reference dictionary
+
+    Returns
+    -------
+    None
+        Does not have a return value. Throws an exception if
+
+    """
+    for key in ref:
+        # if optional is set to true, then dax does not need to have this key
+        if "optional" in ref[key]:
+            if ref[key]["optional"]:
+                continue
+        if not hasattr(dax, key):
+            # Attributes not found in dax
+            raise AttributeError(f"Data array has no attribute '{key}'.")
+
+
+def xr_check_coords(dax: xr.DataArray, ref: dict) -> bool:
+    """Validate the coordinates of the DataArray against a reference dictionary.
+
+    The reference dictionary should have the dimensions as keys and those contain
+    dictionaries with the following keywords (all optional):
+
+    ``values``
+        Specify exact coordinate values to match.
+
+    ``dtype`` : str or type
+        Ensure coordinate dtype matches at least one of the given dtypes.
+
+    ``optional`` : boolean
+        default ``False`` - if ``True``, the dimension has to be in the DataArray dax
+
+    Parameters
+    ----------
+    dax:
+        xarray object which should be validated
+    ref:
+        reference dictionary
+
+    Returns
+    -------
+    bool
+        True, if the test was a success, else throws an exception
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import xarray as xr
+    >>> import weldx as wx
+    >>> dax = xr.DataArray(
+    ...     data=np.ones((3, 2, 3)),
+    ...     dims=["d1", "d2", "d3"],
+    ...     coords={
+    ...         "d1": np.array([-1, 0, 2], dtype=int),
+    ...         "d2": pd.DatetimeIndex(["2020-05-01", "2020-05-03"]),
+    ...         "d3": ["x", "y", "z"],
+    ...     }
+    ... )
+    >>> ref = dict(
+    ...     d1={"optional": True, "values": np.array([-1, 0, 2], dtype=int)},
+    ...     d2={
+    ...         "values": pd.DatetimeIndex(["2020-05-01", "2020-05-03"]),
+    ...         "dtype": ["datetime64[ns]", "timedelta64[ns]"],
+    ...     },
+    ...     d3={"values": ["x", "y", "z"], "dtype": "<U1"},
+    ... )
+    >>> wx.utility.xr_check_coords(dax, ref)
+    True
+
+    """
+    # check if the keys in ref are also in dax
+    xr_valid_key(dax, ref)
```

Comment:
The function is called `xr_check_coords` but it works not only on coordinates but on the whole DataArray.

There should be something like `coords = dax.coords` and only work with `coords` after